src/commands/chain.rs

```diff
@@ -218,9 +228,28 @@ impl Command {
                 let confirmed_tokens = l2_provider.get_confirmed_tokens(from, limit).await?;
                 println!("Confirmed Tokens: {confirmed_tokens:#?}");
             }
-            Command::L1BatchDetails { batch } => {
-                let l1_batch_details = l2_provider.get_l1_batch_details(batch).await?;
-                println!("{l1_batch_details:#?}");
+            Command::L1BatchDetails {
+                batches,
+                proof_time,
+            } => {
+                let current_batch = l2_provider.get_l1_batch_number().await?.as_u32().into();
+
+                let batches_vec = if let Some(batches_vec) = batches {
+                    batches_vec
+                } else {
+                    vec![current_batch]
+                };
+
+                if proof_time {
+                    display_batches_proof_time_from_l1_batch_details(
+                        batches_vec,
+                        current_batch,
+                        l2_provider,
+                    )
+                    .await?;
+                } else {
+                    display_batches_details(batches_vec, current_batch, l2_provider).await?;
+                }
```

Comment:
If you agree with the above these should be the new changes (untested).

```suggestion
            Command::L1BatchDetails {
                mut batches,
                proof_time,
            } => {
                let current_batch = l2_provider.get_l1_batch_number().await?.as_u32().into();

                if batches.is_empty() {
                    batches.push(current_batch);
                }

                if proof_time {
                    display_batches_proof_time_from_l1_batch_details(
                        batches,
                        current_batch,
                        l2_provider,
                    )
                    .await?;
                } else {
                    display_batches_details(batches, current_batch, l2_provider).await?;
                }
```